controllers/users.js

```diff
@@ -284,6 +291,144 @@ class Users {
       }))
       .catch(err => next(err));
   }
+
+  /**
+   * A user can follow another user
+   * @param {object} req
+   * @param {object} res
+   * @return {object} res
+   * */
+  static async follow(req, res) {
+    const { username } = req.params;
+    const { id } = req.user;
+    try {
+      const userExists = await User.findOne({
+        where: {
+          username,
+          id: { [Sequelize.Op.ne]: id },
+        },
+        attributes: ['id', 'username', 'bio', 'image', 'followers']
+      });
+
+      if (!userExists) {
+        return res.status(404).json({
+          status: res.statusCode,
+          message: 'the user you want to follow doesn\'t exist',
+        });
+      }
+
+      const user = await User.findOne({ where: { id }, attributes: ['following'] });
+
+      //  check if the user is already followed
+      const followingUsers = JSON.parse(user.dataValues.following).ids;
+      const existInFollowing = followingUsers.find(followID => followID === userExists.id);
+      if (existInFollowing) {
+        return res.status(400).json({
+          status: 400,
+          message: 'user already followed',
+        });
+      }
+
+      // add a new user to the list of followed users
+      followingUsers.push(userExists.dataValues.id);
+      await User.update(
+        { following: JSON.stringify({ ids: followingUsers }) },
+        { where: { id } }
+      );
+
+      // update the followed user's followers list
+      const followedUsersFollowers = JSON.parse(userExists.dataValues.followers).ids;
+      followedUsersFollowers.push(id);
+      await User.update({
+        followers: JSON.stringify({ ids: followedUsersFollowers })
+      }, {
+        where: { username },
+      });
+
+      return res.status(201).json({
+        status: res.statusCode,
+        profile: {
+          username: userExists.username,
+          image: userExists.image,
+          bio: userExists.bio,
+          following: true,
+        }
+      });
+    } catch (e) {
+      return res.status(500).json({
+        error: e,
+        status: res.statusCode,
+        message: 'something went wrong on the server'
```

Comment:
@avpaul please begin all your messages and error message with a capital letter for example: instead of 'something went wrong on the server', I think it should be 'Something went wrong on the server'

Reply:
Correct!